### ISSUE TITLE
关于loss函数中的labels

### ISSUE THREAD

**Issue by cyy-1234** [chenpan0615/FCCDN_pytorch#9]
您好，我想问一下，在计算loss的时候，labels[1]是怎么得到的，是直接将变化检测的label，宽和高缩小2倍吗，如果是的话，是用的什么策略缩小的（像双线性插值那样吗？），因为我看到第16-19行判断labels[1]==1，我有点迷糊，方便解答一下吗，谢谢了。

```

1. def FCCDN_loss_BCD(scores, labels):
2. # scores = model(input)
3. # labels = [binary_cd_labels, binary_cd_labels_downsampled2times]
4. """ for binary change detection task"""
5. criterion = dice_bce_loss()
6. loss_change = torch.tensor(0.0).cuda()
7. loss_aux = torch.tensor(0.0).cuda()
8. # change loss
9. loss_change = criterion(scores[0], labels[0])
10. # seg map
11. out1 = `torch.sigmoid(scores[1]).clone()
12. out2 = torch.sigmoid(scores[2]).clone()
13. out3 = out1.clone()
14. out4 = out2.clone()
15. 
16. out1[labels[1]==1]=0
17. out2[labels[1]==1]=0
18. out3[labels[1]!=1]=0
19. out4[labels[1]!=1]=0
20. 
21. pred_seg_pre_tmp1 = torch.ones(out1.shape).cuda()
22. pred_seg_pre_tmp1[out1<=0.5]=0
23. pred_seg_post_tmp1 = torch.ones(out2.shape).cuda()
24. pred_seg_post_tmp1[out2<=0.5]=0
25. 
26. pred_seg_pre_tmp2 = torch.ones(scores[1].shape).cuda()
27. pred_seg_pre_tmp2[out3<=0.5]=0
28. pred_seg_post_tmp2 = torch.ones(scores[2].shape).cuda()
29. pred_seg_post_tmp2[out4<=0.5]=0
30. 
31. # seg loss
32. loss_aux = 0.2*criterion(out1, pred_seg_post_tmp1, False)
33. loss_aux += 0.2*criterion(out2, pred_seg_pre_tmp1, False)
34. loss_aux += 0.2*criterion(out3, labels[1]-pred_seg_post_tmp2, False)
35. loss_aux += 0.2*criterion(out4, labels[1]-pred_seg_pre_tmp2, False)
36. 
37. loss = loss_change + loss_aux
38. return loss

```

**Comment by chenpan0615** [comment]
是的，labels的构建过程大概就是下面这样
labels = [change_mask]
avgpool = torch.nn.AvgPool2d((2, 2))
labels.append(avgpool(change_mask))

**Comment by cyy-1234** [comment]
> 是的，labels的制造过程大概就是下面这样 labels = [change_mask] avgpool = torch.nn.AvgPool2d((2, 2)) labels.append(avgpool(change_mask))
好的，谢谢
